File: src/ProjectR.Scripting/content/scripts/spells/OmegaSlash.lua
function GetTargetType()
    return Single
end

function GetName()
  return "Omega Slash"
end

function GetDescription()
    return "Attempts to execute the target.\nLess effective against enemies with high DEF.\nDeals triple damage to targets below 25%% HP."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .88
end

function GetDelay()
  return .1
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  local bonusDamage = 1

  if d:GetCurrentHP() / dHP < .25 then
    bonusDamage = 3
  end

  local damage = (11.25 * aAD - 1.5 * dDEF) * bonusDamage
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/OmegaSlash.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class OmegaSlash : SpellScriptBase
    {
        public override string Name { get { return "Omega Slash"; } }
        public override string Description { get { return "Attempts to execute the target.\nLess effective against enemies with high DEF.\nDeals triple damage to targets below 25%% HP."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 88; } }
        public override double Delay { get { return .1; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var bonusDamage = target.CurrentHP / tHP < .25 ? 3 : 1;

            var damage = (11.25 * cAD - 1.5 * tDEF) * bonusDamage;
            DealDamage(damage);
        }
    }
}